include/merlin_hashtable.cuh

```diff
@@ -934,8 +939,9 @@ class HashTable {
     }
 
     {
+      // Ensure exclusive table access.
```

Comment:
First need to acquire mutex, so that others cannot queue new operations anymore, then make sure any pending asynchronous device operations are finished (i.e., the device is idle).